### PR TITLE
GSoC '20: Samples for Notebook to Kubeflow Deployment using TensorFlow 2.0 Keras

### PR DESCRIPTION
**Pull request by yashjakhotiya**
This is the work product of GSoC '20 project: [Create samples for notebook to Kubeflow deployment using TensorFlow 2.0 Keras](https://summerofcode.withgoogle.com/projects/#5507335985823744)

### CODE REVIEW

**Comment by yashjakhotiya** [comment]
cc: @terrytangyuan @gaocegege @ChanYiLin

**Comment by gaocegege** [comment]
Thanks! I will have a look.

File: tensorflow_2/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Samples for Notebook to Kubeflow Deployment using TensorFlow 2.0 Keras
+This directory aims at building Kubeflow use case samples with Tensorflow 2.0 Keras training code demonstrating 'Customer User Journey' (CUJ) in the process.
```
Comment (terrytangyuan) [inline tensorflow_2/README.md:2]: Add link to CUJ

File: tensorflow_2/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+The samples hosted are as follows -
```
Comment (terrytangyuan) [inline tensorflow_2/README.md:4]: Use complete sentences and paragraphs

File: tensorflow_2/README.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+4. `fairing-with-python-sdk.ipynb` - Jupyter Notebook which deploys a model training task on cloud using [Kubeflow Fairing](https://www.kubeflow.org/docs/fairing/fairing-overview/). 
+Fairing does not require you to build a Docker image of the training code first. Hence, its training code resides in the same notebook.
+
+5. `katib-with-python-sdk.ipynb` - Jupyter Notebook which launches [Katib](https://www.kubeflow.org/docs/components/hyperparameter-tuning/hyperparameter/) 
+hyperparameter tuning experiments using its [Python SDK](https://github.com/kubeflow/katib/tree/master/sdk/python). Katib requires you to build and host a
+Docker image of your training code in a container registry. This sample has used `gcloud builds` to build the required Docker image of the training code along with
+the training data and hosts it on [gcr.io](gcr.io).
+
+6. `tfjob-with-python-sdk.ipynb` - Jupyter Notebook which launches a [TFJob](https://www.kubeflow.org/docs/components/training/tftraining/). TFJobs are used
```
Comment (terrytangyuan) [inline tensorflow_2/README.md:22]: Remove "backend"

File: tensorflow_2/README.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
+# Samples for Notebook to Kubeflow Deployment using TensorFlow 2.0 Keras
+This directory aims at building Kubeflow use case samples with Tensorflow 2.0 Keras training code demonstrating 'Customer User Journey' (CUJ) in the process.
+
+The samples hosted are as follows - 
+### Text Classification 
+Based on a [Tensorflow tutorial](https://www.tensorflow.org/tutorials/keras/text_classification_with_hub), this machine learning task aims at classifying IMDB movie 
+reviews. Building on this, this sample hosts the following,
```
Comment (terrytangyuan) [inline tensorflow_2/README.md:7]: What does "Building on this" mean? Also same comments with writing style

File: tensorflow_2/README.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
+# Samples for Notebook to Kubeflow Deployment using TensorFlow 2.0 Keras
+This directory aims at building Kubeflow use case samples with Tensorflow 2.0 Keras training code demonstrating 'Customer User Journey' (CUJ) in the process.
+
```
Comment (terrytangyuan) [inline tensorflow_2/README.md:3]: Should add prerequistes, system requirements, etc.

File: tensorflow_2/README.md (hunk 1 above)
Comment (yashjakhotiya) [reply tensorflow_2/README.md:2]: Can you please elaborate on what you mean by 'link to CUJ'? Do you want me to put a link to a documentation which explains 'CUJ'?

File: tensorflow_2/README.md (hunk 4 above)
Comment (yashjakhotiya) [reply tensorflow_2/README.md:7]: "Building on this" means we took tensorflow tutorial code as our training code and then demonstrated Kubeflow functionalities on it

File: tensorflow_2/README.md (hunk 1 above)
Comment (terrytangyuan) [reply tensorflow_2/README.md:2]: I thought there would be a slide that demonstrates the CUJ but don't worry about it if not.

File: tensorflow_2/README.md (hunk 4 above)
Comment (terrytangyuan) [reply tensorflow_2/README.md:7]: Probably need to rephrase it so it reads better in proper English.

File: tensorflow_2/README.md (hunk 2 above)
Comment (yashjakhotiya) [reply tensorflow_2/README.md]: I have updated README. Now it uses complete sentences.

File: tensorflow_2/README.md (hunk 3 above)
Comment (yashjakhotiya) [reply tensorflow_2/README.md]: Removed "backend". Just kept it at "Kubernetes".

File: tensorflow_2/README.md (hunk 4 above)
Comment (yashjakhotiya) [reply tensorflow_2/README.md]: Rephrased all notebook details to be more readable.

File: tensorflow_2/README.md (hunk 5 above)
Comment (yashjakhotiya) [reply tensorflow_2/README.md:3]: Mentioned the minimum requirements and gave pointers to more details.

**Comment by jlewi** [comment]
@terrytangyuan Can you please create a separate PR creating the directory with an OWNERs file listing the gsoc mentors as approvers so that they can approve this PR?

**Comment by terrytangyuan** [comment]
@jlewi Done in https://github.com/kubeflow/examples/pull/818.

**Comment by yashjakhotiya** [comment]
@terrytangyuan Updated my directory name to match the directory you created in #818.

**Pull request merged**